src/main/java/io/github/jsonSnapshot/SnapshotFile.java

```diff
@@ -66,8 +68,13 @@ public void push(String snapshot) {
       e.printStackTrace();
     }
 
-    try (FileOutputStream fileStream = new FileOutputStream(file, false)) {
-      byte[] myBytes = StringUtils.join(rawSnapshots, SPLIT_STRING).getBytes();
+    final ArrayList<String> rawItems = new ArrayList<>();
+    for (final SnapshotDataItem snapshotDataItem : storedSnapshots.getItems()) {
+      rawItems.add(snapshotDataItem.asRawData());
+    }
+
+    try (final FileOutputStream fileStream = new FileOutputStream(file, false)) {
+      byte[] myBytes = StringUtils.join(rawItems, SPLIT_STRING).getBytes();
```

Comment:
`myBytes` could be constructed outside of try-catch block in a single stream-based expression.

Reply:
done